### PR TITLE
Adding new models

### PR DESCRIPTION
**Pull request by dakisa11**
Updating model folder with new files

### CODE REVIEW

File: src/main/java/com/study_buddy/study_buddy/model/GroupMemberID.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
+package com.study_buddy.study_buddy.model;
+
+import jakarta.persistence.Embeddable;
+import java.io.Serializable;
+import java.util.Objects;
+
+@Embeddable
+public class GroupMemberID implements Serializable {
+    private Long memberId;
+    private Long groupId;
```
Comment (Mila1308) [inline src/main/java/com/study_buddy/study_buddy/model/GroupMemberID.java:10]: Add proper annotations

File: src/main/java/com/study_buddy/study_buddy/model/GroupMemberID.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        GroupMemberID that = (GroupMemberID) o;
+        return Objects.equals(memberId, that.memberId) && Objects.equals(groupId, that.groupId);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(memberId, groupId);
+    }
```
Comment (Mila1308) [inline src/main/java/com/study_buddy/study_buddy/model/GroupMemberID.java:44]: Remove equals and hashCode. Will add this later

File: src/main/java/com/study_buddy/study_buddy/model/GroupMembers.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    @MapsId("memberId")
+    @ManyToOne
+    @JoinColumn(name = "member_id")
+    private User member;
+
+    @MapsId("groupId")
+    @ManyToOne
+    @JoinColumn(name = "group_id")
+    private StudyGroup group;
+
```
Comment (Mila1308) [inline src/main/java/com/study_buddy/study_buddy/model/GroupMembers.java:21]: Add proper annotations, coordinate object and column names.

File: src/main/java/com/study_buddy/study_buddy/model/LessonParticipantID.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
+package com.study_buddy.study_buddy.model;
+
+import jakarta.persistence.Embeddable;
+import java.io.Serializable;
+import java.util.Objects;
+
+@Embeddable
+public class LessonParticipantID implements Serializable {
+    private Long userId;
+    private Long lessonId;
```
Comment (Mila1308) [inline src/main/java/com/study_buddy/study_buddy/model/LessonParticipantID.java:10]: Add annotations

File: src/main/java/com/study_buddy/study_buddy/model/LessonParticipantID.java [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        LessonParticipantID that = (LessonParticipantID) o;
+        return Objects.equals(userId, that.userId) && Objects.equals(lessonId, that.lessonId);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(userId, lessonId);
```
Comment (Mila1308) [inline src/main/java/com/study_buddy/study_buddy/model/LessonParticipantID.java:41]: Remove this for now

File: src/main/java/com/study_buddy/study_buddy/model/GroupMemberID.java (hunk 1 above)
Comment (savipauk) [reply src/main/java/com/study_buddy/study_buddy/model/GroupMemberID.java:10]: See https://github.com/savipauk/study-buddy/blob/e14b057340740dfb4c4da6b7c900070814cba0c5/src/main/java/com/study_buddy/study_buddy/model/StudyGroup.java#L13 for example

**Pull request closed without merging**